pythran/analyses/immediates.py

```diff
@@ -0,0 +1,25 @@
+""" Immediates gathers immediates. For now, only integers within shape are
+considered as immediates """
+
+from pythran.analyses import Aliases
+from pythran.passmanager import NodeAnalysis
+from pythran.utils import pythran_builtin, attr_to_path
+from pythran.tables import MODULES
+
+import gast as ast
+
+_make_shape = pythran_builtin('make_shape')
+
+class Immediates(NodeAnalysis):
+    def __init__(self):
+        self.result = set()
+        super(Immediates, self).__init__(Aliases)
+
+    def visit_Call(self, node):
+        func_aliases = self.aliases[node.func]
+        for fa in func_aliases:
+            if fa is _make_shape:
```

Comment:
this is unfortunately incorrect: what if (very unlikely) this func alias to make_shape *and* something else? the correct check would be ``if len(func_aliases) == 1 and next(iter(func_aliases)) is _make_shape``

Reply:
Okay gotit, fixed!